test/utils_test.js

```diff
@@ -1068,13 +1074,11 @@ describe("utils", () => {
         $ref: "#/definitions/testdef",
       };
 
-      expect(toIdSchema(schema, undefined, schema.definitions, {}, "rjsf")).eql(
-        {
-          $id: "rjsf",
-          foo: { $id: "rjsf_foo" },
-          bar: { $id: "rjsf_bar" },
-        }
-      );
+      expect(toIdSchema(schema, "rjsf", schema.definitions, {})).eql({
+        $id: "rjsf",
+        foo: { $id: "rjsf_foo" },
+        bar: { $id: "rjsf_bar" },
+      });
```

Comment:
Previously `ui:rootFieldId` on the root element was not passed on to children; now I think it is. Is that intentional? This test almost seems like it was meant to ensure that behavior, but it's a bit hard to say for sure.

Assuming we don't care about preserving the `idPrefix` behavior, this section is called `"should handle idPrefix parameter"`, but there is no longer such a parameter, so maybe we should delete the whole thing.

Reply:
This code here in the master branch
https://github.com/mozilla-services/react-jsonschema-form/blob/7c252de331cfa77c415adf49c04eeb25fd91d28b/src/utils.js#L621-L630
where `id=uiSchema["ui:rootFieldId"]` as shown bellow.
https://github.com/mozilla-services/react-jsonschema-form/blob/7c252de331cfa77c415adf49c04eeb25fd91d28b/src/components/Form.js#L65-L71
So, no matter what, the fact is they are doing the same thing.

### About passing down
Previously, `toIdSchema` didn't pass down to array I guess it's because it relied on array data, which was not given to `toIdSchema` in the very beginning. But now, since the `toIdSchema` has the form data, it should be able to generate a well formed idSchema at the root instead of being called again in `ObjectField` and `ArrayField`.

### About keeping idPrefix
I didn't remove `idPrefix` for compatibility reason because it's documented and someone might have been using it. If we decide to clean it up, it's fine to delete the whole thing.

Reply:
Wait, now I'm even more confused. I definitely misread the code at first, and you're right, it seems like `ui:rootFieldId` and `idPrefix` are doing the same thing, but we're not maintaining compatibility with `idPrefix` -- any `idPrefix` is going to be ignored completely.

I'm the one who merged the PR to add `idPrefix`. I should have paid more attention! But it's definitely documented and I bet @edi9999 uses it somewhere. We should either make an effort to maintain backwards compatibility or admit defeat and add it to #805. What do you think?

Reply:
Yes, I am indeed using idPrefix to uniquely identify form elements from our tests. It could be good if you keep backwards compatibility, but I would understand if you don't.

Reply:
@edi9999 Is there a reason you need `idPrefix` instead of putting `rootFieldId` in the `uiSchema`?